src/workerd/server/server.c++

```diff
@@ -1565,7 +1569,51 @@ private:
 
   kj::Promise<void> writeLogfwdr(uint channel,
       kj::FunctionParam<void(capnp::AnyPointer::Builder)> buildMessage) override {
-    KJ_FAIL_REQUIRE("no logging channels");
+    auto& context = IoContext::current();
+
+    auto headers = kj::HttpHeaders(context.getHeaderTable());
+    auto client = context.getHttpClient(channel, true, nullptr, "writeLogfwdr"_kjc);
+
+    kj::Url url;
+    url.scheme = kj::str("https");
+    url.host = kj::str("fake-host");
+    auto urlStr = url.toString(kj::Url::Context::HTTP_PROXY_REQUEST);
+ 
+    capnp::MallocMessageBuilder requestMessage;
+    auto requestBuilder = requestMessage.initRoot<capnp::AnyPointer>();
+  
+    buildMessage(requestBuilder);
+    capnp::JsonCodec json;
+    auto requestJson = json.encode(requestBuilder.getAs<api::AnalyticsEngineEvent>());
+
+    return context.waitForOutputLocks().then([client = kj::mv(client), urlStr = kj::mv(urlStr),
+        requestJson = kj::mv(requestJson), headers = kj::mv(headers)]() mutable {
+      auto innerReq = client->request(kj::HttpMethod::POST, urlStr, headers, requestJson.size());
+
+      struct RefcountedWrapper: public kj::Refcounted {
+        explicit RefcountedWrapper(kj::Own<kj::HttpClient> client): client(kj::mv(client)) {}
+        kj::Own<kj::HttpClient> client;
+      };
+      auto rcClient = kj::refcounted<RefcountedWrapper>(kj::mv(client));
+      auto req = attachToRequest(kj::mv(innerReq), kj::mv(rcClient));
+
+      return req.body->write(requestJson.begin(), requestJson.size())
+          .attach(kj::mv(requestJson), kj::mv(req.body))
+          .then([resp = kj::mv(req.response)]() mutable {
+        return resp.then([](kj::HttpClient::Response&& response) mutable {
+          if (response.statusCode < 200 || response.statusCode >= 300) {
```

Comment:
Could simplify this using `KJ_REQUIRE(response.statusCode >= 200 && response.statusCode < 300, ...)`